src/object/function.cpp

```diff
@@ -485,16 +487,15 @@ void function::add_to_namespace(
         if (new_func->name().is_none())
             new_func->m_name = name;
 
+        assert(!PyErr_Occurred());
```

Comment:
I don't quite understand: this code here is conditional on `existing` being non-null. So the only way to be here is with the previous `PyObject_GetItem()` call not having failed. So why the `PyErr_Clear()` call between that and here ?

Reply:
Why is it conditional on `existing`? It is outside `if (existing)`.

Reply:
oups, sorry, I misread (the diff view collapsed a few important lines here :-) ). OK, makes sense, then.